core/src/test/java/org/apache/iceberg/TestTableMetadata.java

```diff
@@ -272,4 +273,194 @@ public static String toJsonWithoutSpecList(TableMetadata metadata) {
     }
     return writer.toString();
   }
+
+  @Test
+  public void testJsonWithPreviousMetadataLog() throws Exception {
```

Comment:
This should also check that metadata files without the new content can still be loaded without causing an error. You can use `toJsonWithoutSpecList` or similar methods to generate the older content, then add assertions that validate the parser produces the default value you expect.

Reply:
`testBackwardCompat` kind of takes care of it, I will add assertions for previous metadata.